README.md

```diff
@@ -33,13 +53,107 @@ This archetype assumes a directory structure as follows:
 ```
 .gitignore
 package.json
+tsconfig.json
 
 lib/
   index.js
 
+dist/
+  index.js
+  index.d.js
+  index.js.map
+
+src/
+  index.ts
+
 test/
   spec/**
     *.spec.js
+    *.spec.ts
+```
+
+If you are writing JavaScript that node.js can execute directly, then you can put them in `lib` dir.
+
+If you are using TypeScript, then you can put your `ts` source in `src` dir, and then run `npx tsc` to compile them into the `dist` dir.
+
+`.d.ts` type definition files and source map files will also be generated into the `dist` dir.
+
+## Developing
+
+Once you start writing your code, either as [TypeScript] in `src` or JavaScript in `lib`, you should put your tests in the directory `test/spec` as `*.spec.js` or `*.spec.ts` files.
```

Comment:
does `[TypeScript]` need an associated `(...)` clause?

Reply:
it's at the bottom of the file.